recipes/lora_finetune_distributed.py

```diff
@@ -270,6 +278,124 @@ def setup(self, cfg: DictConfig) -> None:
             last_epoch=self.global_step - 1,
         )
 
+        # Set up profiler, returns FakeProfiler (nullcontext object with no-op `step` method)
+        # if cfg is missing profiler key or if `cfg.profiler.enabled = False`
+        self._profiler = self._setup_profiler(cfg.get(PROFILER_KEY, None), log_cfg=True)
+
+    def _setup_profiler(
+        self, cfg_profiler: DictConfig, log_cfg: bool = False
+    ) -> torch.profiler.profile | FakeProfiler:
+        """
+        Parses the ``profiler`` section of top-level ``cfg`` and sets up profiler
+
+        Args:
+            cfg_profiler: DictConfig - ``profiler`` section of the top-level ``cfg`` (the main config passed to ``recipe.main``)
+            log_cfg: bool - whether to return the profiler config after profiler setup, which sets defaults and possibly
+            overrides certain profiling options.
+
+            NOTE: Since not all settings of the profiler can be parsed from the returned profiler object,
+            such as the ``schedule``, ``log_cfg`` can be used for easy logging / debugging of all profiler options post setup.
+
+        Returns:
+            profiler: torch.profiler.profile | FakeProfiler - FakeProfiler is a nullcontext with no-op methods
+                for ``start``, ``stop``, and ``step`` that can be used in place of ``torch.profiler.profile``
+                if profiler is not enabled such that the instrumented training loop does not need to be changed
+                profiling is disabled.
+            profiler_cfg: Optional[DictConfig]
+
+        The profiler config can be provided in configs under the ``profiler`` key with the following layout:
+
+        .. code-block:: yaml
+            profiler:
+                enabled: bool
+
+                #Output directory of trace artifacts
+                output_dir: str
+
+                #torch.profiler.ProfilerActivity types to trace
+                CPU: bool
+                CUDA: bool
+
+                #Trace options
+                profile_memory: bool
+                with_stack: bool
+                record_shapes: bool
+                with_flops: bool
+
+                #torch.profiler.schedule args
+                schedule:
+                    wait: int
+                    warmup: int
+                    active: int
+                    repeat: int
+
+        """
+
+        enabled = cfg_profiler is not None and cfg_profiler.get("enabled", True)
+
+        if not enabled:
```

Comment:
The logs can be moved into the util as well. I think we'd want the util to take "enabled" as a kwarg as well so we can  have it output the FakeProfiler if it's disabled or on the wrong rank.